internal/crypto/cert.go

```diff
@@ -35,120 +28,6 @@ func CNFromDeviceFingerprint(fingerprint string) (string, error) {
-func MakeSelfSignedCA(certFile, keyFile, serialFile, subjectName string, expiryDays int) (*CA, error) {
-	caConfig, err := makeSelfSignedCAConfig(
-		pkix.Name{CommonName: subjectName},
-		time.Duration(expiryDays)*24*time.Hour,
-	)
-	if err != nil {
-		return nil, err
-	}
-	if err = caConfig.WriteCertConfigFile(certFile, keyFile); err != nil {
-		return nil, err
-	}
-
-	var serialGenerator oscrypto.SerialGenerator
-	if len(serialFile) > 0 {
-		// create / overwrite the serial file with a zero padded hex value (ending in a newline to have a valid file)
-		if err := os.WriteFile(serialFile, []byte("00\n"), 0600); err != nil {
-			return nil, err
-		}
-		serialGenerator, err = oscrypto.NewSerialFileGenerator(serialFile)
-		if err != nil {
-			return nil, err
-		}
-	} else {
-		serialGenerator = &oscrypto.RandomSerialGenerator{}
-	}
-
-	config := TLSCertificateConfig(*caConfig)
-	return &CA{
-		SerialGenerator: serialGenerator,
-		Config:          &config,
-	}, nil
-}
-
-func makeSelfSignedCAConfig(subject pkix.Name, caLifetime time.Duration) (*oscrypto.TLSCertificateConfig, error) {
-	rootcaPublicKey, rootcaPrivateKey, publicKeyHash, err := NewKeyPairWithHash()
-	if err != nil {
-		return nil, err
-	}
-
-	now := time.Now()
-	rootcaTemplate := &x509.Certificate{
-		Subject: subject,
-
-		SignatureAlgorithm: x509.ECDSAWithSHA256,
-
-		NotBefore: now.Add(-1 * time.Second),
-		NotAfter:  now.Add(caLifetime),
-
-		SerialNumber: randomSerial(),
-
-		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
-		BasicConstraintsValid: true,
-		IsCA:                  true,
-
-		AuthorityKeyId: publicKeyHash,
-		SubjectKeyId:   publicKeyHash,
-	}
-	rootcaCert, err := signCertificate(rootcaTemplate, rootcaPublicKey, rootcaTemplate, rootcaPrivateKey)
-	if err != nil {
-		return nil, err
-	}
-	caConfig := &oscrypto.TLSCertificateConfig{
-		Certs: []*x509.Certificate{rootcaCert},
-		Key:   rootcaPrivateKey,
-	}
-	return caConfig, nil
-}
-
-func signCertificate(template *x509.Certificate, requestKey crypto.PublicKey, issuer *x509.Certificate, issuerKey crypto.PrivateKey) (*x509.Certificate, error) {
-	derBytes, err := x509.CreateCertificate(rand.Reader, template, issuer, requestKey, issuerKey)
-	if err != nil {
-		return nil, err
-	}
-	certs, err := x509.ParseCertificates(derBytes)
-	if err != nil {
-		return nil, err
-	}
-	if len(certs) != 1 {
-		return nil, errors.New("expected a single certificate")
-	}
-	return certs[0], nil
-}
-
-// func (ca *CA) EnsureSubCA(certFile, keyFile, serialFile, name string, expireDays int) (*CA, bool, error) {
-// 	if subCA, err := GetCA(certFile, keyFile, serialFile); err == nil {
-// 		return subCA, false, err
-// 	}
-// 	subCA, err := ca.MakeAndWriteSubCA(certFile, keyFile, serialFile, name, expireDays)
-// 	return subCA, true, err
-// }
-
 func (ca *CA) EnsureServerCertificate(certFile, keyFile string, hostnames []string, expireDays int) (*TLSCertificateConfig, bool, error) {
```

Comment:
Is there a reason this method wasn't transferred to the `internal_ca` file? Since the `CA` struct moved it seems like its methods should all be moved too.